neo/Consensus/ConsensusService.cs

```diff
@@ -537,8 +538,13 @@ private void RequestChangeView()
             byte expectedView = context.ChangeViewPayloads[context.MyIndex]?.GetDeserializedMessage<ChangeView>().NewViewNumber ?? 0;
             if (expectedView < context.ViewNumber) expectedView = context.ViewNumber;
             expectedView++;
-            Log($"request change view: height={context.BlockIndex} view={context.ViewNumber} nv={expectedView}");
             ChangeTimer(TimeSpan.FromSeconds(Blockchain.SecondsPerBlock << (expectedView + 1)));
+            if ((context.CountCommitted() + context.CountFailed()) > context.F())
```

Comment:
This is a nice safeguard; if this condition is true though, it should not be possible to actually receive `M` change views, since the number of remaining nodes left to send them should be `< M`, but this helps in the case that the node that is failed and has lost network comes back when it was about to change view and right at the moment others were going to commit since they thought the node had been lost.

Reply:
That's true. We put now a `ping` message  ChangeView(0) to make sure nodes keep talking to each other (and become non-faulty).